src/game/shell.c

```diff
@@ -194,20 +270,41 @@ bool Shell_MakeScreenshot()
         break;
     }
 
-    char path[20];
-    for (int i = 0; i < 10000; i++) {
-        sprintf(path, "screenshot%04d.%s", i, ext);
-        char *full_path = NULL;
-        File_GetFullPath(path, &full_path);
+    // Screenshot folder
+    char *ss_folder_path = NULL;
+    File_GetFullPath("screenshots", &ss_folder_path);
+    File_CreateDirectory(ss_folder_path);
+
+    // Screenshot name
+    char *ss_name =
+        Memory_Alloc(TIMESTAMP_SIZE + LEVEL_TITLE_SIZE + EXTENSION_SIZE);
+    Shell_GetScreenshotName(ss_name);
+
+    // Screenshot folder/name path
+    char *path = Memory_Alloc(strlen(ss_folder_path) + strlen(ss_name));
+    sprintf(path, "%s/%s.%s", ss_folder_path, ss_name, ext);
```

Comment:
Now you're writing beyond the memory variable size. We need to allocate more bytes:
- `+strlen(ext)`
- `+1` for `/`
- `+1` for `.`
- `+1` for the `\0` string terminator
- `+1` for the underscore (the additional number)
- `+3` for maximum of 999 (the additional number)

This gives us `Memory_Alloc(strlen(ss_folder_path) + strlen(ss_name) + 7)`.

Since we're using timestamps the maximum number can be reduced from 1000 to 100 because we will never exceeds 60 screenshots per second. This assumption will reduce the number above by 1 down to `Memory_Alloc(strlen(ss_folder_path) + strlen(ss_name) + 6)`.

Consider doing these `Memory_FreePointer` once like so:

```c
bool result = false;
if (!File_Exists(path)) {
    result = Output_MakeScreenshot(path);
    goto cleanup;
}

for (int i = 2; i < 1000; i++) {
    sprintf(...);
    if (!File_Exists(path)) {
        result = Output_MakeScreenshot(path);
        goto cleanup;
    }
}

cleanup:
Memory_FreePointer(…);
Memory_FreePointer(…);
Memory_FreePointer(…);
return result;
```

Reply:
Ya I had the extension size defined as 20 to account for all that. Is it better to do that, or magic number the `+ 6`?

And yes definitely need to clean up the freeing. I don't personally like goto, so I'm going to push up a slightly different approach.